staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -234,7 +235,7 @@ func (d *dialerCreator) createDialer() utilnet.DialFunc {
 			egressmetrics.Metrics.ObserveDialFailure(d.options.protocol, d.options.transport, egressmetrics.StageConnect)
 			return nil, err
 		}
-		conn, err := proxier.proxy(addr)
+		conn, err := proxier.proxy(ctx, addr)
```

Comment:
we should also plumb ctx to the `proxier, err := d.connector.connect()` call on line 233 so the request lifetime informs how long we wait for a dial

Reply:
correct, tho proxier is only updated out of necessity. The goal was just to update the singleUseGrpcTunnel in this change